merlin-server/sql/merlin/tables/activity.sql

```diff
@@ -83,3 +95,41 @@ create trigger increment_revision_on_delete_activity_trigger
 after delete on activity
 for each row
 execute function increment_revision_on_delete_activity();
+
+create or replace function generate_activity_name()
+returns trigger
+security definer
+language plpgsql as $$begin
+  execute 'create sequence if not exists activity_name_' || new.plan_id || '_' || new.type;
+  new.name = new.type || ' ' || nextval('activity_name_' || new.plan_id || '_' || new.type);
+  return new;
+end$$;
+
+comment on function generate_activity_name() is e''
+  'Generates a name for an activity as the activity type + number from sequence of the type scoped to its plan.';
+
+create trigger generate_name_trigger
+before insert on activity
+for each row execute function generate_activity_name();
+
+comment on trigger generate_name_trigger on activity is e''
+  'Generates a name for an activity as the activity type + number from sequence of the type scoped to its plan.';
+
+create or replace function activity_set_updated_at()
+  returns trigger
+  security definer
+  language plpgsql as $$begin
+  new.last_modified_at = now();
+  return new;
+end$$;
+
+comment on function activity_set_updated_at() is e''
+  'Sets the last_modified_at field of an activity to the current time.';
+
+create trigger set_timestamp
+  before update on activity
+  for each row
+execute procedure activity_set_updated_at();
```

Comment:
Just for my own understanding - what's the difference between `execute function` and `execute procedure`, and why do we use one for `generate_activity_name`, and the other for `activity_set_updated_at`?

Reply:
From https://www.postgresql.org/docs/current/sql-createtrigger.html

```
In the syntax of CREATE TRIGGER, the keywords FUNCTION and PROCEDURE are equivalent, but the referenced function must in any case be a function, not a procedure. The use of the keyword PROCEDURE here is historical and deprecated.

```

Reply:
I went ahead and added a commit normalizing our codebase to using `execute function`

Reply:
Sounds good, thanks!